skema/text_reading/text_reading/grounding/src/main/scala/org/ml4ai/grounding/MiraEmbeddingsGrounder.scala

```diff
@@ -52,14 +58,19 @@ class MiraEmbeddingsGrounder(groundingConcepts:Seq[GroundingConcept], embeddings
     val topKIndices = sortedIndices.take(k)
     val topSimilarities = sortedCosineSimilarities.take(k)
     val topConcepts = topKIndices.map(groundingConcepts)
+
     (topConcepts zip topSimilarities) map {
       case (concept, similarity) => GroundingCandidate(concept, similarity)
     }
   }
 
-  def getNormalizedDistance(text:String, name:String): Float = {
+  private val editDistancesCache = mutable.HashMap[(String, String), Float]()
```

Comment:
Cool!  Is there any danger of the cache getting too big?  I notice signs of a webapp that might be running a long time.

Reply:
Yes, it is possible, although unlikely at this point because the service is being used ephemerally. 

I want us to use an LRU cache instead of a vanilla map. Probably something like this:
https://stackoverflow.com/questions/14561845/lrucache-in-scala